modules/internal/String.chpl

```diff
@@ -577,10 +585,19 @@ module String {
       }
     }
 
+    /*
+      Iterates over the string byte by byte.
+    */
+    iter bytes(): uint(8) {
+      for i in 1..this.len {
```

Comment:
Why wouldn't you write this like this:
```
for i in 0..#this.len {
  yield this.buff[i];
}
```
?

It'd be nice not to create the temporary string on the way. (But there might be an issue with multilocale this way?)

Reply:
Maybe it could be like
``` chapel
var localThis: string = this.localize();
for i in 0..#localThis.len {
  yield localThis.buff[i];
}
```

?

Reply:
It's now better for multilocale.  Thanks.